### DISCUSSION TITLE
Misunderstanding when I should use defaultValue on Controller

### DISCUSSION THREAD

**Discussion by Aaronius** [react-hook-form/react-hook-form#5048] [Q&A] [answered]
The [documentation](https://react-hook-form.com/api/usecontroller/controller) for Controller's `defaultValue` prop states:

> You need to either set defaultValue at the field-level or call useForm with defaultValues.
> If your form will invoke reset with default values, you will need to call useForm with defaultValues instead of setting the defaultValue on individual fields.

In my case, I'm setting `defaultValues` when I call `useForm` and I also plan on calling `reset` with default values. Therefore, according to instructions, I'm not setting `defaultValue` at the individual Controller level.

I'm leveraging `useFieldArray` to show a list of rows. When I delete a row, then add a row with new data, the row shows stale data. Code speaks a thousand words, so here's my sandbox:

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-q3uxl?file=/src/index.js

Notice when I click the DELETE button on the existing row, then click the APPEND button, I would expect the newly added row to say `appendLuo`, but instead it's `Luo`, which was from the row I just deleted.

When I set `defaultValue` at the Controller level, I see the functionality I would expect, but that seems to contradict what the documentation is telling me I should do.

What am I misunderstanding?

Also, if it turns out I _am_ supposed to set `defaultValue` on the Controller, this makes the API a bit weird (at least it seems like it to me) when I'm building wrappers to 3rd-party components to make them compatible with React-Hook-Form. For example:

https://codesandbox.io/s/react-hook-form-usefieldarray-with-defaultvalue-vxzf5?file=/src/index.js

Look at ThirdPartyComponentWrapper. Would you have the parent (ItemList) pass `defaultValue` as a prop into `ThirdPartyComponentWrapper`? What's the standard way of doing something like this?

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the feedback. I have just updated the doc to leave a note when used with `useFieldArray`. You probably have awarded, useFieldArray is uncontrolled and doesn't trigger re-render while you are typing in the input, hence we will need defaultValue provided during user shuffle and move around the inputs. 

For the external component, you can provide the value down as `prop` and assigned to `Controller/useController`'s defaultValue.

eg

```ts
<ThirdPartyComponentWrapper defaultValue={fireld.firstName} />

const ThirdPartyComponentWrapper = ({ defaultValue }) => {
  return <Controller defaultValue={defaultValue} />
}
```
